extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/TraceMultiPropagator.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.extensions.trace.propagation;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A propagator designed to inject and extract multiple trace {@code HttpTextFormat} propagators,
+ * intendended for backwards compatibility with existing services using different formats. It works
+ * in a stack-fashion, starting with the last registered propagator, to the first one.
+ *
+ * <p>Upon injection, this propagator invokes {@code HttpTextFormat#inject()} for every registered
+ * trace propagator. This will result in the carrier containing all the registered formats.
+ *
+ * <p>Upon extraction, this propagator invokes {@code HttpTextFormat#extract()} for every registered
+ * trace propagator, returning immediately when a successful extraction happened.
+ *
+ * <pre>{@code
+ * HttpTextFormat traceFormats = TraceMultiPropagator.builder()
+ *   .addPropagator(new MyCustomTracePropagator())
+ *   .addPropagator(new JaegerPropagator())
+ *   .addPropagator(new HttpTraceContext())
+ *   .build();
+ * // Register it in the global propagators:
+ * OpenTelemetry.setPropagators(
+ *     DefaultContextPropagators.builder()
+ *       .addHttpTextFormat(traceFormats)
+ *       .build());
+ * ...
+ * // Extraction will be performed in reverse  order, i.e. starting with the last
+ * // registered propagator (HttpTraceContext in this example).
+ * Context context = OpenTelemetry.getPropagators().getHttpTextFormat()
+ *   .extract(context, carrier, carrierGetter);
+ * }</pre>
+ *
+ * @since 0.6.0
+ */
+@Immutable
+public class TraceMultiPropagator implements HttpTextFormat {
+  private final HttpTextFormat[] propagators;
+  private final List<String> propagatorsFields;
+
+  private TraceMultiPropagator(List<HttpTextFormat> propagatorList) {
+    this.propagators = new HttpTextFormat[propagatorList.size()];
+    propagatorList.toArray(this.propagators);
+
+    List<String> fields = new ArrayList<>();
```

Comment:
Should we validate if fields are duped? Or does it not matter? Mainly wondering whether it's ok for this list to have dupes.

Reply:
Still not clear how to proceed here (we have a similar case for the base `CompositePropagator` in context-prop). I will fill an issue for that before merging this PR.

(My gut feeling is that we should de-dup them at ctor time).

Reply:
If you can add a TODO to the issue here would be nice

Reply:
An issue is even better ;) https://github.com/open-telemetry/opentelemetry-java/issues/1388

Reply:
Meant it'd be good to link to that issue here with a code TODO - reduces chance of the issue just sitting in the backlog if it distracts in the code ;)